src/coreComponents/python/modules/geosx_xml_tools_package/geosx_xml_tools/main.py

```diff
@@ -0,0 +1,66 @@
+"""Command line tools for geosx_xml_tools"""
+
+import sys
+import argparse
+from geosx_xml_tools import xml_processor
+
+
+def parse_arguments():
+    # Parse the user arguments
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-i', '--input', type=str, action='append', help='Input file name (multiple allowed)')
+    parser.add_argument('-c', '--compiled-name', type=str, help='Compiled xml file name (otherwise, it is randomly genrated)', default='')
+    parser.add_argument('-s', '--schema', type=str, help='GEOSX schema to use for validation', default='')
+    parser.add_argument('-v', '--verbose', type=int, help='Verbosity of outputs', default=0)
+    parser.add_argument('-p', '--parameters', nargs='+', action='append', help='Parameter overrides (name value, multiple allowed)', default=[])
+    return parser.parse_known_args()
+
+
+def preprocess_serial():
+    """
+    Entry point for the geosx_xml_tools console script
+    """
+    # Process the xml file
+    args, unknown_args = parse_arguments()
+    compiled_name = xml_processor.process(args.input,
+                                          outputFile=args.compiled_name,
+                                          schema=args.schema,
+                                          verbose=args.verbose,
+                                          parameter_override=args.parameters)
+    return format_geosx_arguments(compiled_name, unknown_args)
+
+
+def preprocess_parallel():
+    """
+    MPI aware xml preprocesing
+    """
+    # Process the xml file
+    from mpi4py import MPI
+    comm = MPI.COMM_WORLD
+    rank = comm.Get_rank()
```

Comment:
We need to have separate serial and parallel (called using pygeosx) methods because using ``mpi4py`` seems to break the MPI environment for subsequent calls to GEOSX.  I haven't been able to track down why this is happening unfortunately.